test/uninsured-vesting-v1/uninsured-vesting-v1.test.ts

```diff
@@ -220,89 +221,172 @@ describe("VestingV1", () => {
       });
 
       it("handles zero token balance gracefully", async () => {
         const startingBalance = await someOtherToken.methods.balanceOf(vesting.options.address).call();
         expect(startingBalance).to.be.bignumber.zero;
-        await vesting.methods.recoverToken(someOtherToken.options.address).send({ from: deployer });
+        await vesting.methods.recoverToken(someOtherToken.options.address).send({ from: daoWallet });
         expect(await someOtherToken.methods.balanceOf(vesting.options.address).call()).to.be.bignumber.zero;
       });
     });
 
     describe("access control", () => {
-      it("cannot call activate if not owner", async () => {
-        await expectRevert(async () => vesting.methods.activate(await getDefaultStartTime()).send({ from: anyUser }), "Ownable: caller is not the owner");
+      const generateAccessControlErrorMsg = (account: string, role: string) =>
+        `VM Exception while processing transaction: reverted with reason string 'AccessControl: account ${account.toLowerCase()} is missing role ${role}'`;
+
+      describe("only project", () => {
+        it("can activate", async () => {
+          const projectRole = await vesting.methods.PROJECT_ROLE().call();
+
+          const expectedInvalidUsers = [daoWallet, deployer];
+
+          for (const invalidUser of expectedInvalidUsers) {
+            await expectRevert(
+              async () => vesting.methods.activate(await getDefaultStartTime()).send({ from: invalidUser }),
+              generateAccessControlErrorMsg(invalidUser, projectRole)
+            );
+          }
+
+          await vesting.methods.setAmount(user1, await projectToken.amount(TOKENS_PER_USER)).send({ from: projectWallet });
+          await vesting.methods.activate(await getDefaultStartTime()).send({ from: projectWallet });
+          // should not revert
+        });
+
+        it("can set amounts", async () => {
+          const projectRole = await vesting.methods.PROJECT_ROLE().call();
+
+          const expectedInvalidUsers = [anyUser, daoWallet, deployer];
+
+          for (const invalidUser of expectedInvalidUsers) {
+            await expectRevert(
+              async () => vesting.methods.setAmount(user1, 1).send({ from: invalidUser }),
+              generateAccessControlErrorMsg(invalidUser, projectRole)
+            );
+          }
+
+          await vesting.methods.setAmount(user1, 1).send({ from: projectWallet });
+          // should not revert
+        });
+
+        it("can emergency claim", async () => {
+          await setAmountForUser1();
+          await approveProjectTokenToVesting();
+          await activateAndReachStartTime();
+          await vesting.methods.emergencyRelease().send({ from: daoWallet });
+          await expectRevert(async () => vesting.methods.emergencyClaim(user1).send({ from: anyUser }), Error.OnlyProjectOrSender);
+
+          await vesting.methods.emergencyClaim(user1).send({ from: projectWallet });
```

Comment:
just making sure - do we have a test checking the user1 can also emergencyClaim?